sample: syst: Add config for testing on intel_adsp_cavs platforms

peter-mitsis opened the pull request:
Duplicated branch solution (see PR #44232). The original author is on vacation and I am unable to push updates to the original PR due to deviations caused by a rebase).

Original description below ...

Adding configuration for enabling this test on cavs platforms.
This patch also contains a fix to mipi_syst library which resolves
the memory alignment issue across different architectures.

Fixes #43344

aasthagr committed:
sample: syst: Add intel_adsp_cavs support

Adding configurations for enabling this sample project on cavs
platforms. This patch also contains a fix to mipi_syst library
which resolves the memory alignment issue across different
architectures.

Fixes #43344

Signed-off-by: Aastha Grover <aastha.grover@intel.com>
Signed-off-by: Peter Mitsis <peter.mitsis@intel.com>

dcpleung commented on west.yml:
```diff
@@ -182,7 +182,7 @@ manifest:
       groups:
         - debug
-      revision: 2f51407b2ac68e2be41af322a42905063f009d75
+      revision: pull/9/head
```
Need to remove this here. The west update commit should then be moved before this commit.

peter-mitsis replied:
That is addressed in the next commit in the set, but I can squash them.

dcpleung replied:
You can squash them, or reorder so the west.yml change appears beforehand. Using a PR as west module revision is simply for CI to test the whole thing before things are merged in the module repo. Once the changes are in the module repo, west.yml should use SHA instead.

nashif commented on samples/subsys/logging/syst/sample.yaml:
```diff
@@ -60,3 +64,16 @@ tests:
         - "SYS-T RAW DATA: "
     extra_configs:
       - CONFIG_LOG_MIPI_SYST_USE_CATALOG=y
+  sample.logger.syst.adsp_cavs:
```
dont understand why you need this entry and why it is being made specific to this platform, isnt this already covered above? You need to remove the platform_allow from all other scenarios and make all scenarios runnable on all platforms.

peter-mitsis replied:
My understanding has been that when using logging with an adsp_cavs board such as intel_adsp_cavs18, it uses the adsp backend instead of the uart backend, which has necessitated some changes to the .conf files.  I'll dig some more into this.

peter-mitsis replied:
I think I understand more now than I did before. If I have the right mental model, then the sample.logger.syst.adsp_cavs entry is not needed as the custom .conf files under boards/ provide the necessary overrides.

peter-mitsis replied:
Here's what I am finding ... there are various configurations that syst does not support. These include at least 64-bit and big-endian. There may be more. This may be a bit of after-the-fact rationalisation, but I think the original use of platform_allowed instead of integration_platforms may have been done to address those limitations with a simple set of rules.

peter-mitsis committed:
samples: syst: Update sample.yaml

Updates the sample.yaml file to exclude problem architectures
from this sample project. SYS-T does not support 64-bit or
big endian architectures. The posix arch is excluded as it
is generating numerous anomolous runtime messages because it
does have the means to determine when data resides in the
rodata section.

It also converts the list of allowed platforms to lists of
platforms to use for integration testing. This helps to increase
general testing coverage.

Signed-off-by: Peter Mitsis <peter.mitsis@intel.com>

MaureenHelm commented on samples/subsys/logging/syst/sample.yaml:
```diff
@@ -1,9 +1,19 @@
+    integration_platforms:
+      - mps2_an385
+      - qemu_x86
+      - qemu_x86_64
```
Is this a mistake, since you're filtering out 64-bit platforms?

peter-mitsis replied:
Good catch.

yerabolu reviewed (commented):
@peter-mitsis: Tried running the sample on intel_adsp_cavs25 and these 2 tests fail
sample.logger.syst.v1.deferred
sample.logger.syst.v1.immediate

```
DEBUG   - DEVICE: [00000001] <dbg> syst: mixed str dynamic str --- dynamic str --- another dynamic str --- another dynamic str
DEBUG   - DEVICE: [00000001] <dbg> syst: mixed c/s ! static str dynamic str static str !
DEBUG   - DEVICE: [00000001] <err> syst: frame
DEBUG   - DEVICE: 03 d2 04 00 00 08 01 02  03 04 05 06 07 08       |........ ......
DEBUG   - DEVICE: [00000001] <wrn> syst: frame
DEBUG   - DEVICE: 03 d2 04 00 00 08 01 02  03 04 05 06 07 08       |........ ......
DEBUG   - DEVICE: [00000001] <inf> syst: frame
DEBUG   - DEVICE: 03 d2 04 00 00 08 01 02  03 04 05 06 07 08       |........ ......
DEBUG   - DEVICE: [00000002] <dbg> syst: frame
DEBUG   - DEVICE: 03 d2 04 00 00 08 01 02  03 04 05 06 07 08       |........ ......
DEBUG   - DEVICE: hello sys-t on board intel_adsp_cavs25
```
and it is waiting for the device to become available.

peter-mitsis committed:
logging: Enable Sys-T on ADSP backend

Enables the Sys-T formatting on the ADSP backend when the
SYST logging format is selected.

Signed-off-by: Peter Mitsis <peter.mitsis@intel.com>
